File: programmers/43165.cpp
#include <string>
#include <vector>

using namespace std;

int answer =0;

void DFS(vector<int> &num, int &t, int depth, int sum){
    if(depth == num.size()){
        if(sum == t){
            answer++;
        }
    }
    else{
        DFS(num, t, depth+1, sum + num[depth]);
        DFS(num, t, depth+1, sum - num[depth]);
    }
}

int solution(vector<int> numbers, int target) {
    
    DFS(numbers, target, 0, 0);
    
    return answer;
}

File: Javascript/programmers/43165.js
function solution(numbers, target) {
  let answer = 0;
  DFS(0, 0);
  function DFS(l, sum) {
    if (l === numbers.length) {
      if (sum === target) answer++;
    } else {
      DFS(l + 1, sum + numbers[l]);
      DFS(l + 1, sum - numbers[l]);
    }
  }

  return answer;
}
